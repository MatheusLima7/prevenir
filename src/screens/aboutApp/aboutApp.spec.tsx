import { render } from "@testing-library/react-native";

import AboutApp from "./aboutApp";

jest.mock("react-native-linear-gradient", () => jest.fn());

jest.mock("@react-native-vector-icons/FontAwesome6", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("@react-native-vector-icons/evil-icons", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe("About App  Screen Component", () => {
  it("should render component correctly", () => {
    const container = render(<AboutApp />);

    expect(container).toMatchSnapshot();
  });
});
