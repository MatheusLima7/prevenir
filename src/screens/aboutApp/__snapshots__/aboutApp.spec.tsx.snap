// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About App  Screen Component should render component correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#ECFEFF",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 50,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 25,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 5,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "shadowColor": "#000000",
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="always"
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={true}
      style={{}}
    >
      <View>
        <Text
          color="gray"
          size="medium"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-SemiBold",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 30,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
          testId="text-component"
          type="title"
        >
          Sobre este APP
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
          testId="text-component"
          type="text"
        >
          "Bem-vindo ao Espaço Prevenir, uma iniciativa do Governo do Estado de São Paulo para apoiar pessoas que sofrem pelo transtorno por uso de substâncias."
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-SemiBold",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 20,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
          testId="text-component"
          type="title"
        >
          Nossa Missão
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
          testId="text-component"
          type="text"
        >
          Nosso objetivo é fornecer uma ferramenta acessível e eficaz para auxiliar no processo de recuperação e reinserção social de pessoas que sofrem pelo transtorno de uso de substâncias.
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-SemiBold",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 20,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
          testId="text-component"
          type="title"
        >
          Recursos do APP
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              {},
            ]
          }
          testId="text-component"
          type="text"
        >
          ●  
          Informações sobre serviços de apoio no estado de São Paulo.
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              {},
            ]
          }
          testId="text-component"
          type="text"
        >
          ●  
          Linha direta de apoio 24/7
        </Text>
        <Text
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              {},
            ]
          }
          testId="text-component"
          type="text"
        >
          ●  
          Acompanhamento personalizado do progresso.
        </Text>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginBottom": 60,
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <Text
      color="gray"
      size="xMinSmall"
      style={
        [
          {
            "color": "#717171",
            "display": "flex",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 10,
            "fontWeight": "400",
            "lineHeight": 30,
          },
          {},
        ]
      }
      testId="text-component"
      type="text"
    >
      Desenvolvido pela
    </Text>
    <Image
      source={
        {
          "testUri": "../../../src/assets/10i9.png",
        }
      }
      style={
        {
          "marginLeft": 7,
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
